### PR TITLE
[29588] Add fail2ban to xTAU

### PR DESCRIPTION
**Pull request by andersdd**
Adds fail2ban install and basic configuration

Sets basic config file only - most likely needs more work on the nginx/postfix configuration which is held in the fail2ban.sh script

### CODE REVIEW

File: fail2ban.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+$POSTFIX
+
+[postgresql-iptables]
+enabled  = true
+filter   = postgresql
+action   = iptables[name=postgresql-iptables, port=$PGPORT, protocol=udp]
+           sendmail-whois[name=Postgresql, dest=dest=cloudops@xtuple.com]
+logpath  = /var/log/postgresql/*.log
+EOF
```
Comment (gilmoskowitz) [inline fail2ban.sh]: Again consider using a template. The `PGPORT` replacement should already be handled. Perhaps the template handling functions could/should be expanded to handle additional, single-use replacements like `NGINX` and `POSTFIX` here.

File: fail2ban.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            <HOST> FATAL:  pg_hba.conf rejects connection for host .+$
+            <HOST> FATAL:  unsupported frontend protocol .+$
+ignoreregex = : Successful su for (postgres) by root$
+              New session \d+ of user (postgres)\.$
+              Removed session \d+\.$
+EOF
+
+POSTFIX=""
+  if type "postfix" > /dev/null 2>&1 ; then
+read -d '' POSTFIX <<EOF
```
Comment (gilmoskowitz) [inline fail2ban.sh]: The `read` shouldn't be necessary:
```bash
POSTFIX="[postfix]
enabled = true
...
backend = %(postfix_backend)s"
```

File: fail2ban.sh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+configure_fail2ban() {
+  echo "In: ${BASH_SOURCE} ${FUNCNAME[0]} $@"
+
+  F2B=`pgrep -f fail2ban`
+  if [ -z $F2B ]; then
+    install_fail2ban
+  fi
+
+  if [ -z "$PGPORT" ] && [ "$MODE" = "manual" ]; then
+    PGPORT=$(whiptail --backtitle "$( window_title )" --inputbox "Enter the PostgreSQL database post number" 8 60 3>&1 1>&2 2>&3)
```
[the remark this answers is not in the document]
Comment (andersdd) [reply fail2ban.sh]: fixed.

**Comment by andersdd** [comment]
Moved fail2ban configuration to template files.  Let's see if you like my in-template variable replacement method

**Pull request merged**